perfkitbenchmarker/providers/ibmcloud/ibmcloud_virtual_machine.py

```diff
@@ -0,0 +1,599 @@
+# Copyright 2020 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Class to represent a IBM Cloud Virtual Machine. """
+
+import base64
+import json
+import logging
+import threading
+import os
+import time
+
+from absl import flags
+from perfkitbenchmarker import disk
+from perfkitbenchmarker import virtual_machine
+from perfkitbenchmarker import linux_virtual_machine
+from perfkitbenchmarker import windows_virtual_machine
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import providers
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.providers.ibmcloud import ibm_api as ibm
+from perfkitbenchmarker.providers.ibmcloud import ibmcloud_disk
+from perfkitbenchmarker.providers.ibmcloud import ibmcloud_network
+from perfkitbenchmarker.providers.ibmcloud import util
+
+FLAGS = flags.FLAGS
+
+_DEFAULT_VOLUME_IOPS = 3000
+_WAIT_TIME_RHEL = 300
+_WAIT_TIME_UBUNTU = 600
+_WAIT_TIME_DEBIAN = 120
+
+
+class IbmCloudVirtualMachine(virtual_machine.BaseVirtualMachine):
+  """Object representing a IBM Cloud Virtual Machine"""
+
+  CLOUD = providers.IBMCLOUD
+  IMAGE_NAME_PREFIX = None
+
+  _lock = threading.Lock()
+  command_works = False
```

Comment:
It looks like "command_works" is a check to see if a Login needs to be done, is that right?  If so I would build that into the IbmCloudApi() mechanism to do a login if needed (1st time using it or if the last attempt responded with an unauthorized error)

Reply:
that is correct, however it is really not needed, it's been removed.